pkg/utils/config.go

```diff
@@ -16,87 +19,386 @@ type Credential struct {
 }
 
 type HarborConfig struct {
-	CurrentCredentialName string       `yaml:"current-credential-name"`
-	Credentials           []Credential `yaml:"credentials"`
+	CurrentCredentialName string       `mapstructure:"current-credential-name" yaml:"current-credential-name"`
+	Credentials           []Credential `mapstructure:"credentials" yaml:"credentials"`
+}
+
+var (
+	CurrentHarborData    *HarborData
+	CurrentConfig        *HarborConfig
+	configMutex          sync.RWMutex
+	configInitialization sync.Once
+	configInitError      error
+)
+
+type HarborData struct {
+	ConfigPath string `yaml:"configpath"`
 }
 
 var (
-	HarborFolder      string
+	HarborConfigDir   string
+	HarborDataDir     string
 	DefaultConfigPath string
+	DefaultDataPath   string
 )
 
 func SetLocation() {
 	home, err := os.UserHomeDir()
 	if err != nil {
-		log.Fatal(err)
+		log.Fatalf("Unable to determine user home directory: %v", err)
+	}
+
+	xdgConfigHome := os.Getenv("XDG_CONFIG_HOME")
```

Comment:
I followed the instructions in [this specification](https://specifications.freedesktop.org/basedir-spec/latest/). According to section 3 the xdg home should default to $HOME/.config in case the XDG environment variables are not set in the shell/env. Therefore I thought it could be helpful to check whether a user has a $HOME, since this might not be fulfilled for e.g. machine users.

Reply:
But I will double check whether the result in the expected behavior, and want to include this in some unit testing.